### DISCUSSION TITLE
Having difficulty accessing wagmi context in an external library.

### DISCUSSION THREAD

**Discussion by oveddan** [wevm/wagmi#4373] [Q&A]
I'm trying to write an external library of helper hooks that internally call wagmi's hooks.

For example, this component `useMintable` in my external library `my-react-lib`:

```ts
"use client";

import { useChainId } from "wagmi";

import { useChainId } from '
const useMintable = () => {
   const chainId = useChainId();
}
```

I get this error:
"WagmiProviderNotFoundError: `useConfig` must be used within `WagmiProvider`."
<img width="968" alt="Screenshot 2024-11-01 at 4 08 23 PM" src="https://github.com/user-attachments/assets/c51a029e-f1d7-4f08-bc94-09de49a893d5">

...which doesn't make sense since this hook is called from within the wrapped provider component, as seen below:

In my Next app, which im generating using `pnpm wagmi create`:

layout.tsx:
```tsx
"use client";

import "./globals.css";
import type { Metadata } from "next";
import { Inter } from "next/font/google";
import { headers } from "next/headers";
import { type ReactNode } from "react";
import { cookieToInitialState } from "wagmi";

import { getConfig } from "../wagmi";
import { Providers } from "./providers";
import Account from "./components/Account";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Create Wagmi",
  description: "Generated by create-wagmi",
};

export default function RootLayout(props: { children: ReactNode }) {
  const initialState = cookieToInitialState(
    getConfig(),
    headers().get("cookie")
  );
  return (
    <html lang="en">
      <body className={inter.className}>
        <Providers initialState={initialState}>
          <Account />
          {props.children}
        </Providers>
      </body>
    </html>
  );
}

```

providers.tsx:
```tsx
"use client";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { type ReactNode, useState } from "react";
import { type State, WagmiProvider } from "wagmi";

import { getConfig } from "@/wagmi";

export function Providers(props: {
  children: ReactNode;
  initialState?: State;
}) {
  const [config] = useState(() => getConfig());
  const [queryClient] = useState(() => new QueryClient());

  return (
    <WagmiProvider config={config} initialState={props.initialState}>
      <QueryClientProvider client={queryClient}>
        {props.children}
      </QueryClientProvider>
    </WagmiProvider>
  );
}
```

page.tsx:
```tsx
"use client";

import { useGetMintable } from "my-react-lib";

export default function Mint() {

  useGetMintable()

  return <></>
}
```

**Comment by tmm** [comment]
Usually see this in a monorepo or other environment, where there are multiple Wagmi versions installed. Happy to look into it if there's a minimal reproduction.

**Comment by oveddan** [comment]
@tmm I've been able to repro in a mimimal turborepo example:

https://github.com/oveddan/wagmi-hook-issue-example

See the simple hook here: https://github.com/oveddan/wagmi-hook-issue-example/blob/main/packages/ui/src/useMint.ts
And where it is used here: https://github.com/oveddan/wagmi-hook-issue-example/blob/main/apps/web/src/app/page.tsx#L10

Both the UI package, as well as the web app, are using the same versions of wagmi and viem.

**Comment by Abdullah1738** [comment]
Having this same issue
